src/lib/modes/mode_pad/mode_pad.cpp

```diff
@@ -121,5 +125,44 @@ size_t OneAndZeros_Padding::unpad(const byte block[], size_t size) const
    return (size-1);
    }
 
+/*
+* Pad with ESP Padding Method
+*/
+void ESP_Padding::add_padding(secure_vector<byte>& buffer,
+                                      size_t last_byte_pos,
+                                      size_t block_size) const
+   {
+   byte pad_value = 0x01;
+
+   for(size_t i = last_byte_pos; i < block_size; ++i)
+      {
+      buffer.push_back(pad_value++);
+      }
+   }
+
+/*
+* Unpad with ESP Padding Method
+*/
+size_t ESP_Padding::unpad(const byte block[], size_t size) const
+   {
+   while(size)
```

Comment:
Shouldn't this confirm all the padding bytes? Just truncating at the last found 0x01 seems very dangerous to me - it seems to me this function should both confirm last byte is <= block_size and confirm block[i-1] == block[i] for i >= (block_size - last_byte)

For extra challenge I'm pretty sure this can be done in entirely const time, just loop over the entire last block and ignore the input when i corresponds to plaintext bytes.